### PR TITLE
Support requesting module/process infos

### PR DESCRIPTION
**Pull request by hwbllmnn**
@terrestris/devs Please review.

### CODE REVIEW

**Review by marcjansen — APPROVED** [review]
One suggestion, you decide.

Can we report the coverage to coveralls, please?

Looks good otherwise.

File: src/main/java/de/terrestris/actinia/ActiniaClient.java [hunk 1]
```diff
@@ -133,4 +135,60 @@ public Location getLocation(String name) {
...
+      return new ArrayList<>(this.modules.values());
+    }
+    HttpRequest request;
+    try {
+      request = HttpRequest.newBuilder(new URI(String.format("%slatest/modules", url))).build();
+      ObjectMapper mapper = new ObjectMapper();
+      HttpResponse<InputStream> response = client.send(request, HttpResponse.BodyHandlers.ofInputStream());
+      JsonNode node = mapper.readTree(response.body());
+      if (!node.get("status").asText().equals("success")) {
+        throw new ActiniaException("Getting the locations was unsuccessful.");
```
Comment (metzm) [inline src/main/java/de/terrestris/actinia/ActiniaClient.java:149]:
```suggestion
        throw new ActiniaException("Getting the modules was unsuccessful.");
```

Comment (hwbllmnn) [reply src/main/java/de/terrestris/actinia/ActiniaClient.java]: Nice catch, thanks!

File: src/main/java/de/terrestris/actinia/Module.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  }
+
+  public List<Parameter> getOutputParameters() {
+    if (outputParameters == null) {
+      client.updateDetails(this);
+    }
+    return outputParameters;
+  }
+
+  public void addInputParameter(Parameter parameter) {
```
[the remark this answers is not in the document]
Comment (hwbllmnn) [reply src/main/java/de/terrestris/actinia/Module.java:49]: Let's have another look at the thing when we have a complete picture (expect another PR later!)

**Pull request merged**